#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/robots/gctronic/e-puck/protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/Floor.proto"
EXTERNPROTO "../protos/MazeWallEastWest.proto"
EXTERNPROTO "../protos/MazeWallNorthSouth.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.5773502691896257 0.5773502691896257 0.5773502691896257 2.0943245915045035
  position 0.000767452768272147 0.00017963976745574969 2.5429722886373094
}
TexturedBackground {
}
TexturedBackgroundLight {
  luminosity 1.5
}
Floor {
  size 1 1
}
DEF epuck E-puck {
  translation -0.1249999999999996 -0.3749999998941362 -6.396199783865152e-05
  controller "Lab2_Task2"
  supervisor TRUE
  turretSlot [
    DEF Front DistanceSensor {
      translation 0.045 0 -0.011
      name "front distance sensor"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF Right DistanceSensor {
      translation 0.01 -0.035 -0.011
      rotation 0 0 1 -1.5707853071795865
      name "right distance sensor"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF Rear DistanceSensor {
      translation -0.025 0 -0.011
      rotation 0 0 1 3.1415
      name "rear distance sensor"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF Left DistanceSensor {
      translation 0.01 0.035 -0.011
      rotation 0 0 1 1.5708
      name "left distance sensor"
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    InertialUnit {
      rotation -1 0 0 -5.307179586466759e-06
    }
    Camera {
      translation 0.043 0 -0.015
      name "camera1"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
  ]
}
MazeWallNorthSouth {
  translation -0.375 0.5 0.025
  name "Cell 0 North Wall"
}
MazeWallEastWest {
  translation -0.5 0.375 0.025
  name "Cell 0 West Wall"
}
MazeWallNorthSouth {
  translation -0.125 0.5 0.025
  name "Cell 1 North Wall"
}
MazeWallNorthSouth {
  translation -0.125 0.25 0.025
  name "Cell 1 South Wall"
}
MazeWallNorthSouth {
  translation 0.125 0.5 0.025
  name "Cell 2 North Wall"
}
MazeWallNorthSouth {
  translation 0.125 0.25 0.025
  name "Cell 2 South Wall"
}
MazeWallNorthSouth {
  translation 0.375 0.5 0.025
  name "Cell 3 North Wall"
}
MazeWallEastWest {
  translation 0.5 0.375 0.025
  name "Cell 3 East Wall"
}
MazeWallNorthSouth {
  translation -0.375 0 0.025
  name "Cell 4 South Wall"
}
MazeWallEastWest {
  translation -0.5 0.125 0.025
  name "Cell 4 West Wall"
}
MazeWallNorthSouth {
  translation -0.125 0 0.025
  name "Cell 5 South Wall"
}
MazeWallEastWest {
  translation 0.25 0.125 0.025
  name "Cell 6 East Wall"
}
MazeWallEastWest {
  translation 0.5 0.125 0.025
  name "Cell 7 East Wall"
}
MazeWallEastWest {
  translation -0.5 -0.125 0.025
  name "Cell 8 West Wall"
}
MazeWallNorthSouth {
  translation -0.125 -0.25 0.025
  name "Cell 9 South Wall"
}
MazeWallEastWest {
  translation 0.25 -0.125 0.025
  name "Cell 10 East Wall"
}
MazeWallNorthSouth {
  translation 0.125 -0.25 0.025
  name "Cell 10 South Wall"
}
MazeWallEastWest {
  translation 0.5 -0.125 0.025
  name "Cell 11 East Wall"
}
MazeWallEastWest {
  translation -0.25 -0.375 0.025
  name "Cell 12 East Wall"
}
MazeWallNorthSouth {
  translation -0.375 -0.5 0.025
  name "Cell 12 South Wall"
}
MazeWallEastWest {
  translation -0.5 -0.375 0.025
  name "Cell 12 West Wall"
}
MazeWallNorthSouth {
  translation -0.125 -0.5 0.025
  name "Cell 13 South Wall"
}
MazeWallNorthSouth {
  translation 0.125 -0.5 0.025
  name "Cell 14 South Wall"
}
MazeWallEastWest {
  translation 0.5 -0.375 0.025
  name "Cell 15 East Wall"
}
MazeWallNorthSouth {
  translation 0.375 -0.5 0.025
  name "Cell 15 South Wall"
}
